### PR TITLE
[Script Execution] Create bootstrap checkpoint

### PR DESCRIPTION
**Pull request by sideninja**
This PR creates a bootstrap checkpoint and stores it in a file that will be ingested by the access node on bootstrap to populate the register index database. 
Since the register updates during bootstrap aren't observed by state execution sync API we have agreed this will be the best alternative.

### CODE REVIEW

File: cmd/bootstrap/run/execution_state.go [hunk 1]
```diff
@@ -67,11 +54,38 @@ func GenerateExecutionState(
...
+	root := 0
+	err = checkpointer.Checkpoint(root)
+	if err != nil {
+		return flow.DummyStateCommitment, err
+	}
+
+	trie, err := checkpointer.LoadCheckpoint(root)
+	if err != nil {
+		return flow.DummyStateCommitment, err
+	}
```
Comment (zhangchiqing) [inline cmd/bootstrap/run/execution_state.go:83]: I would suggest we do this instead:
```

tries := ledgerStorage.Tries()
trie := findTrieByStateCommit(tries, stateCommitment)

```

We could implement the `findTrieByStateCommit` in the `ledger.FindTrieByStateCommit`

Comment (sideninja) [reply cmd/bootstrap/run/execution_state.go:83]: so avoid using checkpointer? makes sense, I didn't like hardcoding root at 0 anyway.

Comment (zhangchiqing) [reply cmd/bootstrap/run/execution_state.go:83]: Yeah, but more importantly to validate the statecommitment of the trie.

File: cmd/bootstrap/run/execution_state.go [hunk 2]
```diff
@@ -67,11 +55,30 @@ func GenerateExecutionState(
...
+
+	matchTrie, err := ledgerStorage.FindTrieByStateCommit(stateCommitment)
+	if err != nil {
+		return flow.DummyStateCommitment, err
+	}
+	if matchTrie == nil {
+		return flow.DummyStateCommitment, fmt.Errorf("bootstraping failed to produce a checkpoint")
```
Comment (zhangchiqing) [inline cmd/bootstrap/run/execution_state.go:75]: better to log the state commitment
```suggestion
		return flow.DummyStateCommitment, fmt.Errorf("bootstraping failed to produce a checkpoint for trie %v", stateCommitment)
```

**Review by zhangchiqing — APPROVED** [review]

**Review by peterargue — APPROVED** [review]

**Pull request merged**